contrib/remote-ssh-build.bash

```diff
@@ -0,0 +1,90 @@
+#!/bin/bash
+# vim: softtabstop=2 shiftwidth=2 expandtab
+
+## REMOTE SSH ACCESS
+
+# This script builds a zbm with remote access support via ssh. The image and
+# efi executable will contain private host keys and public keys used for
+# mutual authentication.
+
+# All keys / settings are placed into ${RS_DIR} ('ssh-data' in the current
+# working directory by default). Its contents can be customized as needed.
+# They are regenerated if missing.
+
+# Server private keys are generated automatically as needed. It is
+# recommended to not reuse existing private keys.
+# User authentication for the root account is done via 'authorized_keys'.
+# The current users' authorized keys are copied if it is not present.
+
+RS_DIR=$(realpath "${RS_DIR:-ssh-data}")
+RS_KEYDIR="${RS_DIR}/keys"
+
+mkdir -p "${RS_DIR}"
+mkdir -p "${RS_KEYDIR}"
+
+
+## PREPARE SERVER PRIVATE KEYS
+
+# If no local ssh host keys are available, create them
+# WARNING: These private keys will be part of the final image!
+RS_PK_RSA="${RS_KEYDIR}"/ssh_host_rsa_key
+RS_PK_ECDSA="${RS_KEYDIR}"/ssh_host_ecdsa_key
+RS_KEYGEN_OPT=(-m PEM -N '' -C "zbm-server-key")
+if [ ! -f "${RS_PK_RSA}" ]; then
+  ssh-keygen -t rsa "${RS_KEYGEN_OPT[@]}" -f "${RS_PK_RSA}"
+fi
+if [ ! -f "${RS_PK_ECDSA}" ]; then
+  ssh-keygen -t ecdsa "${RS_KEYGEN_OPT[@]}" -f "${RS_PK_ECDSA}"
+fi
+
+
+## PREPARE AUTHORIZED KEYS
+
+RS_AUTH="${RS_KEYDIR}/authorized_keys"
+if [ ! -f "${RS_AUTH}" ]; then
+  echo "Cannot find ${RS_AUTH}, copying from current user"
+  cp -v ~/.ssh/authorized_keys "${RS_AUTH}"
+fi
+
+
+## PREPARE SETTINGS
+
+RS_DNC="${RS_DIR}/dracut-network.conf"
```

Comment:
If the entire `cmdline.d` is mounted rather than a single-file bind, it becomes easy to inject other customized cmdline.d snippets and avoids polluting the build directory with a bunch of files that have different conceptual purposes.

```suggestion
mkdir -p "${RS_DIR}/cmdline.d"
RS_DNC="${RS_DIR}/cmdline.d/dracut-network.conf"
```

Reply:
I was worried that I'd override files created by other stages in the build pipeline.